### PR TITLE
v0_dev_SSR-1580: Create contributing guide

### PR DESCRIPTION
**Pull request by h000780**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,31 +16,10 @@ With the release of v1, we will continue to mainly fix bugs and update libraries
...
-./dist/react/kintone-ui-component.min.js
-./dist/react/kintone-ui-component.min.css
-./esm/constant
-./esm/css
-./esm/react
-./esm/js
-```
+## Contributing
+
+We welcome [contributions](https://github.com/kintone-labs/kintone-ui-component/blob/v0_dev/CONTRIBUTING.md) to kintone UI Component.
```
Comment (TomokoMiyake) [inline README.md:21]:
```suggestion
We welcome contributions to kintone UI Component.  
For the details, please check [CONTRIBUTING.md](https://github.com/kintone-labs/kintone-ui-component/blob/v0_dev/CONTRIBUTING.md).
```

File: README.md [hunk 2]
```diff
@@ -16,31 +16,10 @@ With the release of v1, we will continue to mainly fix bugs and update libraries
...
-./dist/kintone-ui-component.min.js
-./dist/kintone-ui-component.min.css
-./dist/react/kintone-ui-component.min.js
-./dist/react/kintone-ui-component.min.css
-./esm/constant
-./esm/css
-./esm/react
-./esm/js
-```
+## Contributing
```
Comment (TomokoMiyake) [inline README.md:19]:
```suggestion
## Contributing Guide

```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+Contributions are always welcome!  
+If you have discovered a bug or have a feature request, [please create an issue on GitHub](https://github.com/kintone-labs/kintone-ui-component/issues/new/choose).  
+Pull requests are also welcome when you find trivial bugs or typos. 
+
+We use English for all commit messages, code comments, issues, pull requests.  
+
+Please follwing [Note for Naming Rule](#note-for-naming-rule).  
+
+
+This guide to develop and the explanation of the command options.
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:12]:
```suggestion
If you discover a bug or have a feature request, please [create an issue on GitHub](https://github.com/kintone-labs/kintone-ui-component/issues/new/choose).  
Pull requests are also welcome when you find a trivial bug or a typo. 
	
We use English for all commit messages, code comments, issues, pull requests.    
Please follow [Note for Naming Rule](#note-for-naming-rule) when creating an issue or a pull request.  
	
Here is the guide on the command options and how to develop the package.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+## Command Guide
+
+The commands are providing as npm-critpts, please executing `npm run {{ Command }}` in the terminal.
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:16]:
```suggestion
The commands are provided as npm-scripts, so please execute `npm run {Command}` in the terminal.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+|build:esm|package and generate the ECMAScript Modules (ESM)|
+|build:umd|package and generate the production version of the umd file|
+|build:umd:dev|package and generate the development version of the umd file|
+|transpile|compile and build TypeScript code|
+|test|unit test|
+|eslint|eslint check|
+## How to develop
+### Clone and create branch
+
+#### 1.Please clone from GitHub repository at [here](https://github.com/kintone-labs/kintone-ui-component) to local
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:30]:
```suggestion
#### 1. Clone from GitHub repository [here](https://github.com/kintone-labs/kintone-ui-component) to local
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+```sh
+git clone git@github.com:kintone-labs/kintone-ui-component.git
+```
+
+#### 2.Check out to branch for dev
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:36]:
```suggestion
#### 2. Checkout branch for development
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+Please refer [Branch naming rule](#Branch) and create a new branch(base on [branch:v0_dev](https://github.com/kintone-labs/kintone-ui-component/tree/v0_dev/)) for developing.
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:37]:
```suggestion
Please refer to [Branch naming rule](#Branch) and create a new branch(base on [branch:v0_dev](https://github.com/kintone-labs/kintone-ui-component/tree/v0_dev/)) for development from the next version branch. 
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+#### 2.Check out to branch for dev
+Please refer [Branch naming rule](#Branch) and create a new branch(base on [branch:v0_dev](https://github.com/kintone-labs/kintone-ui-component/tree/v0_dev/)) for developing. 
+```sh
+git checkout <branch v0_dev>
+git branch <branch name>
+git checkout <branch name>
+```
+### Setup
+This repository is a Siggle-repo Monolith using npm.
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:44]: What is Siggle-repo Monolith??

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+```sh
+cd kintone-ui-component
+npm ci
+```
+### Develop
+#### 1. Implement your functions/features
+#### 2. Eslint check
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:51]:
```suggestion
#### 2. ESLint check
```sh
npm run eslint
```
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+#### 3. Commit and Create Pull-Request(Naming Rule refer [here](#Pull-Request))
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:52]:
```suggestion
#### 3. Commit and Create Pull Request
Please refer to [Pull Request naming rule](#Pull-Request).
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+### Build
+```sh
+npm run build:umd
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:56]:
```suggestion
npm run build
or
npm run build:umd:dev
or
npm run build:umd
or
npm run build:esm
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+```
+### Update document
+Please refer [Document with MKDocs](https://github.com/kintone-labs/kintone-ui-component/blob/v0_dev/docs/README.md).
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:59]:
```suggestion
Please refer to [Document with MKDocs](https://github.com/kintone-labs/kintone-ui-component/blob/v0_dev/docs/README.md).
```

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+
+
+## Note for Naming Rule
+
+### Branch
+
+You can create many other child branches to implement source code for functions/features and/or unit tests and/or user guides.
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:66]:
```suggestion
You can create several other child branches to implement source code for functions/features and/or unit tests and/or user guides.
```

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+
+The format of the Branch Name is as follows
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:68]:
```suggestion
The format of the Branch Name is as follows:  
```

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+
+## Note for Naming Rule
+
+### Branch
+
+You can create many other child branches to implement source code for functions/features and/or unit tests and/or user guides.
+
+The format of the Branch Name is as follows  
+```sh
+[Version No]_Title
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:70]:
```suggestion
{Version Number}_Title
```

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+```
+For example
+```sh
+origin/master/v1.1.0
+  |__origin/master/v1.1.0_Button_src
+  |__origin/master/v1.1.0_Button_unit_test
+  |__origin/master/v1.1.0_Button_docs
+  |__origin/master/v1.1.0_Notification_docs  
+  |__origin/master/v1.1.0_Text_src
+
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:80]: You can delete the lines for v1.
```suggestion
```

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+origin/v0_dev/v0.8.3
+  |__origin/v0_dev/v0.8.3_Button_src
+  |__origin/v0_dev/v0.8.3_Button_unit_test
+  |__origin/v0_dev/v0.8.3_Button_docs
+  |__origin/v0_dev/v0.8.3_Notification_docs  
+  |__origin/v0_dev/v0.8.3_Text_src
+```
+### Pull Request
+
+If you finish the implementation of the child branches, you should merge from the parent branch before creating a pull request against it.
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:90]:
```suggestion
If you finish the implementation under the child branches, you should merge from the parent branch before creating a pull request against it.
```

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+  |__origin/v0_dev/v0.8.3_Button_unit_test
+  |__origin/v0_dev/v0.8.3_Button_docs
+  |__origin/v0_dev/v0.8.3_Notification_docs  
+  |__origin/v0_dev/v0.8.3_Text_src
+```
+### Pull Request
+
+If you finish the implementation of the child branches, you should merge from the parent branch before creating a pull request against it.  
+
+The format of the PR title is as follows
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:92]:
```suggestion
The format of the Pull Request Title is as follows:  
```

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+  |__origin/v0_dev/v0.8.3_Notification_docs  
+  |__origin/v0_dev/v0.8.3_Text_src
+```
+### Pull Request
+
+If you finish the implementation of the child branches, you should merge from the parent branch before creating a pull request against it.  
+
+The format of the PR title is as follows  
+```sh
+[PR Type] PR Title
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:94]:
```suggestion
[Pull Request Type] Title
```

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+
+PR types are as follows  
+| Type |	Example |
+| :---- | :---- |
+| Bugfix |	[BugFix] Fix a bug |
+| HotFix |	[HotFix] Urgently fix something |
+| CS | [CS] Fix Coding Style |
+| Feature | [Feature] Wrote a new feature |
+| Update | [Update] Changed a doc file |
+| Refactor | [Refactor] Refactored a certain function |
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:105]:
```suggestion
Pull Request types are as follows:  
| Type | Example |
| :---- | :---- |
| BugFix | [BugFix] Fix a bug |
| HotFix | [HotFix] Urgently fix something |
| CS | [CS] Fix coding style |
| Feature | [Feature] Write a new feature |
| Update | [Update] Change a doc file |
| Refactor | [Refactor] Refactor a certain function |
```

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+
+### Issue
+
+The format of the issue title is as follows
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:109]:
```suggestion
The format of the Issue Title is as follows:  
```

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+| CS | [CS] Fix Coding Style |
+| Feature | [Feature] Wrote a new feature |
+| Update | [Update] Changed a doc file |
+| Refactor | [Refactor] Refactored a certain function |
+
+### Issue
+
+The format of the issue title is as follows  
+```sh
+[Issue Type] Issue Title
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:111]:
```suggestion
[Issue Type] Title
```

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
+### Issue
+
+The format of the issue title is as follows  
+```sh
+[Issue Type] Issue Title  
+```
+|Type| Example|
+|:----|:----|
+|Bug report| [Bug report] Found a bug about XXX |
+|Feature request|[Feature request] Suggest an idea about XXX|
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:116]:
```suggestion
|Type| Example|
|:----|:----|
|BugReport| [BugReport] Found a bug about XXX |
|FeatureRequest|[FeatureRequest] Suggest an idea about XXX|
```

**Review by TomokoMiyake — COMMENTED** [review]
In addition to the review comment, please update issue template with the naming rule.

File: CONTRIBUTING.md (hunk 8 above)
Comment (h000780) [reply CONTRIBUTING.md]: This means that there is only one project in this repository.
https://css-tricks.com/from-a-single-repo-to-multi-repos-to-monorepo-to-multi-monorepo/

File: CONTRIBUTING.md [hunk 24]
```diff
@@ -27,57 +27,60 @@ The commands are providing as npm-critpts, please executing `npm run {{ Command
...
+#### 2. CheckOut branch for development
+Please refer to [Branch naming rule](#Branch) and create a new branch(base on [branch:v0_dev](https://github.com/kintone-labs/kintone-ui-component/tree/v0_dev/)) for development from the next version branch. 
 ```sh
 git checkout <branch v0_dev>
 git branch <branch name>
 git checkout <branch name>
 ```
 ### Setup
-This repository is a Siggle-repo Monolith using npm.
+This repository is a Siggle-repo using npm.
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:44]: Please fix the misspelling.
> Single Repo

File: CONTRIBUTING.md [hunk 25]
```diff
@@ -27,57 +27,60 @@ The commands are providing as npm-critpts, please executing `npm run {{ Command
...
 ```sh
 git clone git@github.com:kintone-labs/kintone-ui-component.git
 ```
 
-#### 2.Check out to branch for dev
-Please refer [Branch naming rule](#Branch) and create a new branch(base on [branch:v0_dev](https://github.com/kintone-labs/kintone-ui-component/tree/v0_dev/)) for developing. 
+#### 2. CheckOut branch for development
+Please refer to [Branch naming rule](#Branch) and create a new branch(base on [branch:v0_dev](https://github.com/kintone-labs/kintone-ui-component/tree/v0_dev/)) for development from the next version branch. 
 ```sh
 git checkout <branch v0_dev>
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:39]: Will this be the correct command?
git checkout v0_dev

File: CONTRIBUTING.md [hunk 26]
```diff
@@ -87,30 +90,30 @@ origin/v0_dev/v0.8.3
...
+| HotFix | [HotFix] Urgently fix something |
+| CS | [CS] Fix coding style |
+| Feature | [Feature] Write a new feature |
+| Update | [Update] Change a doc file |
+| Refactor | [Refactor] Refactor a certain function |
 
 ### Issue
 
-The format of the issue title is as follows  
+The format of the Issue Title is as follows
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:112]: Please add `:` at the end of the line.

File: CONTRIBUTING.md [hunk 27]
```diff
@@ -36,12 +36,12 @@ git clone git@github.com:kintone-labs/kintone-ui-component.git
...
 Please refer to [Branch naming rule](#Branch) and create a new branch(base on [branch:v0_dev](https://github.com/kintone-labs/kintone-ui-component/tree/v0_dev/)) for development from the next version branch. 
 ```sh
-git checkout <branch v0_dev>
+git checkout v0_dev
 git branch <branch name>
 git checkout <branch name>
 ```
 ### Setup
-This repository is a Siggle-repo using npm.
+This repository is a Siggle Repo using npm.
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:44]: I was saying about `Single` not `Siggle`.

**Pull request merged**